src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -304,6 +305,22 @@ protected void activate(final BundleContext bundleContext, final ResourceResolve
             this.observationPaths[i] = new Path(paths[i]);
         }
 
+        // optimize alias path allow list
+        this.aliasPathAllowList.clear();
```

Comment:
Hi @akankshajain18, sorry if I might have pointed you in the wrong direction, but now changing the aliasPathAllowList is no longer an atomic operation as @jsedding suggested.

There must be some way to this correctly with the volatile keyword without triggering Sonar errors.

Reply:
Hi @wimsymons
If I use a volatile keyword with any collection, a sonar bug is triggered.
( Non-primitive fields as volatile is not allowed.) 
That's why I used 'AtomicReferenceArray' as suggested by the sonar cloud.
Later @jsedding and everyone in the review suggested using copyonwriteArraylist.

Please let me know if there is an issue with copyonwriteArraylist.

Reply:
Tough. Sonar is wrong.

Reply:
please check my draft build where I have used volatile keyword
https://github.com/apache/sling-org-apache-sling-resourceresolver/pull/34